README.md

```diff
@@ -3,13 +3,51 @@
+There are many reasons for this but most simplyput look at the graphs of different functions:
+![look the pretty lines](image.png)
+some functions start slow and scale up fast, others start fast and end up slow. In the real world you can have much more complex functions aswell. 
+
+2. asymptotic complexity does not factor in external factors such as cashe misses, the temperature of the air, hardware accelarators and so on.
+our program's fisrt run at 1000 elements warmed up the CPU to an extent where it is underclocing to keep itself safe, this only happend because-
+you never clean your computer + your running a 350W threadripper with a stock cooler resting naked held securely by gravity ontop of your cpu. ![my computer before my real cooler arived](image-1.png)
+Now that the scene has been set, your seccond script, this time running a larger test of 10,000 elements, loads into memory. But tragicly it overflows into
+the swapfile! (as we only put 1 stick of 2 gb ram in our system...) The 50mb floppy disk we use as our prefered swapfile location is just enough to 
+pick up the slack and 100 secconds later after HARD but severely throttled number crunching (mostly disk reads) we calculate our final answer... 42!
+
+3. If we always ignore constant factors and mostly care about leading terms, then they can add up and lead to a significant discrepancy in our runtime.
+
 - Suppose finding a particular element in a binary search tree with 1,000
   elements takes 5 seconds. Given what you know about the asymptotic complexity
   of search in a binary search tree, how long would you guess finding the same
   element in a search tree with 10,000 elements takes? Explain your reasoning.
 
+i would say that the worst case would take:
+![6.66](calculator.png)
+6.6666 secconds
+
+reasoning: i set it up like ratio and solved for the missing variable. I assumed that-
+the scaling holdes and that there are no other variables affecting the runtime. and ig that it was the worst case i was calculating.
+
 - You measure the time with 10,000 elements and it takes 100 seconds! List 3
   reasons why this could be the case, given that reasoning with the asymptotic
   complexity suggests a different time.
 
+1. asymptotic complexity does not measure external circumstances such as the computer does not "efficently",
+work with numbers above a certain size, so if all were doing is adding expoentially larger numbers while addition at small input size has hardware-
+accelarators we may have to fall back to a software solution at larger inputs. the specifics were just an example of my thought process but i think you get what im thinking.-
+This does effect small inputs that that is my point, maby 1,000 elements was enough to fit inside the (enter example number here) bit addition curcitry, but the 10,000 didnt-
+so now we have to fall back to the software solution to compute the result.
```

Comment:
This sounds the same as your next reason, i.e. the larger input didn't fit into RAM.